backend/Users/Users.ts

```diff
@@ -71,3 +68,18 @@ export async function findUser(email, password) {
     return null;
   }
 }
+
+export async function findUserByEmail(
+  email: string
+): Promise<UserModel | null> {
+  const users = await sequelize.models.User.findAll({
+    where: {
+      email: Sequelize.where(
+        Sequelize.fn("LOWER", Sequelize.col("email")),
+        Op.eq,
+        email?.toLowerCase()
```

Comment:
It's not valid for the findUserByEmail function to be called without a valid email, for two reasons:

1. Typescript wouldn't allow it due to our strong typing of email in the function signature. It is not optional and null/undefined are not allowed
2. The SQL generated by sequelize would match users who have a null email which is invalid at the database level since (iirc) the email column is NOT NULL. So it would never match anything and going to the DB would be pointless in that case.

Because of this, the optional chaining here is not helpful and should be removed. Instead, it should be email.toLowerCase()

Reply:
Excellent.  This makes sense.  Thank you!